framework/src/components/system/system.js

```diff
@@ -138,6 +140,7 @@ class System {
 				.toString('hex');
 			this.headers.broadhash = newBroadhash;
 			this.logger.debug('System headers', this.headers);
+			this.emit('update', this.headers);
```

Comment:
The current solution seems weird to me. We emit update from the component, listen it on chain and republish again to the channel... I would prefer to publish to the channel directly in the chain module after calling the update() function. That way we avoid the modification of the system component.

Reply:
I considered this approach but the update function was called in multiple places and some of those were too deep in the logic (it wouldn't make sense to pass a channel object there). Also, it's better if all channel actions/events are invoked/published from the top level of the module; it's more readable that way.

Reply:
I see the point. But in that case, I would prefer to publish to the channel straight from the component since this it's not restricted in the LIP.

Reply:
Please, @jondubois check if what you propose in this PR is aligned to https://github.com/LiskHQ/lisk/issues/3036.

Reply:
Not allowing components to publish directly to a channel is an important principle; it's better if components don't know about framework-level concepts.

This problem will be addressed properly by https://github.com/LiskHQ/lisk/issues/3036

Reply:
If that's right, we should update the LIP because it's stating the opposite. 
https://github.com/LiskHQ/lips/blob/master/proposals/lip-0005.md#components
```Component can use channels if required for implementation behavior. ```

Reply:
Yes I think this should be updated. It's related to the discussion we had last week about components managing state. The conclusion was that it would be better if components would not be responsible for managing/maintaining shared state because it breaks the `one source of truth` principle; they should behave more like utilities or clients.